doc/design/proposal-strip-types.md

```diff
@@ -0,0 +1,100 @@
+# Strip Types Design
+
+## Problem
+
+TypeScript support has been for a few years at the top of the list of features that users want to see in Node.js.
+Discussion has been on going for a few years [#43818](https://github.com/nodejs/node/issues/43818), and the main blocker has been the lack of a clear on long term support.
+TypeScript does not follow semver, and the Node.js project has been reluctant to commit to supporting TypeScript in the long term, given that the lifespan of an LTS version is 3 years.
+
+## Proposal
+
+The proposal is to add a new flag to the Node.js CLI, `--experimental-strip-types`, which will replace inline TypeScript types with whitespace.
+No type checking is performed, and types are discarded.
+
+Example:
+
+```typescript
+interface Foo {
+    bar: string;
+}
+
+throw new Error("Whitespacing");
+
+```
+
+Becomes:
+
+```javascript
+
+
+
+
+throw new Error("Whitespacing");
+
+```
+
+>Note that the missing types are replaced with whitespace, so the line numbers are preserved, so that sourcemaps are not needed.
+
+By removing types completely, we can avoid the need to commit to supporting TypeScript in the long term, as the feature is not a full TypeScript implementation. We won't need to worry about breaking changes in TypeScript, as we are not implementing them.
+TypeScript features that cannot be replaced by whitespace, such as enums, will error.
+As in JavaScript files, file extensions are required in `import` statements and `import()` expressions.
+TypeScript features that depend on settings within `tsconfig.json`, such as paths or converting module formats, are unsupported.
+This will solve the long term compatibility issue, but might not be as complete as a full TypeScript implementation.
+
+### Goals
+
+#### Determining module system
```

Comment:
The latter half of this document is just a copy of the docs from the PR, but those docs are likely to continue to change. Rather than having two documents to keep in sync, can we focus this one on the high level and just link to the PR for implementation details?

I think a good thing for this document to cover would be the pros and cons of this approach as compared with others, as that’s something that we wouldn’t discuss in the API docs.